test/analytics.js

```diff
@@ -1,526 +1,145 @@
-      expect(hit.qt).to.be.at.most(100)
-    })
-
-    analytics.ga(DEFAULT_REQUEST_OBJ).pageView()
-  })
-
-  it('session id producer is not called if propertySettings is empty', () => {
-    const sessionIdProducer = sinon.spy(r => {})
-    const analytics = new Analytics({
-      propertySettings: [],
-      sessionIdProducer,
-      attributionProducer: r => {}
-    })
-    analytics.ga(DEFAULT_REQUEST_OBJ).pageView()
-    expect(sessionIdProducer.callCount).to.equal(0)
+  beforeEach(() => {
+    process.env.ANALYTICS_XGOV_PROPERTY = 'testProperty'
+    process.env.ANALYTICS_PROPERTY_API = 'testSecret'
   })
-
-  it('attribution producer is not called if propertySettings is empty', () => {
-    const attributionProducer = sinon.spy(r => {})
-    const analytics = new Analytics({
-      propertySettings: [],
-      sessionIdProducer: r => {},
-      attributionProducer
-    })
-    analytics.ga(DEFAULT_REQUEST_OBJ).pageView()
-    expect(attributionProducer.callCount).to.equal(0)
-  })
-  ;[
-    { desc: 'campaign and source specified', test: { campaign: 'cn1', source: 'cs' }, res: true },
-    { desc: 'campaign and id specified', test: { campaign: 'cn2', id: 'cid2' }, res: true },
-    { desc: 'campaign omitted', test: { source: 'cs', id: 'cid3' }, res: false },
-    { desc: 'source and id omitted', test: { campaign: 'cn1' }, res: false },
-    { desc: 'campaign specified but zero length', test: { campaign: '', source: 'cs' }, res: false },
-    { desc: 'campaign specified but just spaces', test: { campaign: '   ', source: 'cs' }, res: false },
-    { desc: 'source specified but zero length', test: { campaign: 'cn1', source: '' }, res: false },
-    { desc: 'source specified but just spaces', test: { campaign: 'cn1', source: '   ' }, res: false },
-    { desc: 'id specified but zero length', test: { campaign: 'cn1', id: '' }, res: false },
-    { desc: 'id specified but just spaces', test: { campaign: 'cn1', id: '   ' }, res: false }
-  ].forEach(({ desc, test, res }) => {
-    it(`validates that return value of attribution producer has a cn value and one of cm or cs (${desc})`, async () => {
-      const attributionProducer = () => test
-      const analytics = new Analytics({
-        propertySettings: TEST_PROPERTY_SETTINGS,
-        sessionIdProducer: r => {},
-        attributionProducer
-      })
-      sinon.spy(console, 'warn')
-      if (res) {
-        await analytics.ga(DEFAULT_REQUEST_OBJ).pageView()
-        expect(console.warn.calledWith('Attribution should contain campaign and one of source or id')).to.equal(false)
-      } else {
-        await analytics.ga(DEFAULT_REQUEST_OBJ).pageView()
-        expect(console.warn.calledWith('Attribution should contain campaign and one of source or id')).to.equal(true)
-      }
-    })
+  afterEach(() => {
+    delete process.env.ANALYTICS_XGOV_PROPERTY
+    delete process.env.ANALYTICS_PROPERTY_API
+    delete process.env.HTTPS_PROXY
+    delete process.env.https_proxy
```

Comment:
hapi-gapi shouldn't be relying on environment variables, it should be set by config. If the client then wants to use environment variables, it can, but that decision should lie there not in the plugin.